fission/common.go

```diff
@@ -168,31 +168,14 @@ func createArchive(client *client.Client, fileName string, specFile string) *fis
 		archive.Type = fission.ArchiveTypeLiteral
 		archive.Literal = contents
 	} else {
-		// make a kubernetes client
-		_, kubeClient, _, err := crd.GetKubernetesClient()
-		if err != nil {
-			fatal(err.Error())
-		}
-
-		fissionNamespace := os.Getenv("FISSION_NAMESPACE")
-
-		// get svc end point for storagesvc
-		service, err := kubeClient.CoreV1().Services(fissionNamespace).Get("storagesvc", metav1.GetOptions{})
-		if err != nil {
-			fatal(fmt.Sprintf("Error getting storage service object from kubernetes :%v", err.Error()))
-		}
-
 		u := strings.TrimSuffix(client.Url, "/") + "/proxy/storage"
 		ssClient := storageSvcClient.MakeClient(u)
 
 		// TODO add a progress bar
 		id, err := ssClient.Upload(fileName, nil)
 		checkErr(err, fmt.Sprintf("upload file %v", fileName))
 
-		// this needs to be storagesvc.fission
-		storageSvcEndpoint := fmt.Sprintf("http://%s.%s/", service.Name, service.Namespace)
-		storageServiceClient := storageSvcClient.MakeClient(storageSvcEndpoint)
-		archiveUrl := storageServiceClient.GetUrl(id)
+		archiveUrl := ssClient.GetUrl(id)
```

Comment:
This will work if FISSION_URL or server flag is set. But if both of them are not set, the CLI will set up local port forwarding to the controller. In this case, the curl will be `127.0.0.1:<localpost>`, and make the archive URL point to a local address, not the storage service URL. I think that's the main reason why CLI trying to get storagesvc service object from k8s API server to make an internal archive URL.

Reply:
Oh right, good catch, I checked the package I created and indeed it has a URL which points to `127.0.0.1` - which won't work. I will re-work and update the PR.